scripts/flaky_tests.py

```diff
@@ -0,0 +1,90 @@
+#! /usr/bin/env python3
+
+import argparse
+import json
+import logging
+from collections import defaultdict
+from typing import DefaultDict
+
+import psycopg2
+import psycopg2.extras
+
+# We call the test "flaky" if it changed its status on the same revision in the last N=10 days on main branch.
+FLAKY_TESTS_QUERY = """
+    SELECT
+        DISTINCT ON (parent_suite, suite, name) parent_suite, suite, name
+    FROM
+        (
+            SELECT
+                revision,
+                jsonb_array_elements(data -> 'children') -> 'name' as parent_suite,
+                jsonb_array_elements(jsonb_array_elements(data -> 'children') -> 'children') -> 'name' as suite,
+                jsonb_array_elements(jsonb_array_elements(jsonb_array_elements(data -> 'children') -> 'children') -> 'children') -> 'name' as name,
+                jsonb_array_elements(jsonb_array_elements(jsonb_array_elements(data -> 'children') -> 'children') -> 'children') -> 'status' as status,
+                to_timestamp((jsonb_array_elements(jsonb_array_elements(jsonb_array_elements(data -> 'children') -> 'children') -> 'children') -> 'time' -> 'start')::bigint / 1000)::date as timestamp
+            FROM
+                regress_test_results
+            WHERE
+                reference = 'refs/heads/main'
+        ) data
+    WHERE
+        timestamp > CURRENT_DATE - INTERVAL '%s' day
+    GROUP BY
+        revision, parent_suite, suite, name
+    HAVING
+        COUNT(DISTINCT status) > 1
+    ;
+"""
+
+
+def main(args: argparse.Namespace):
+    connstr = args.connstr
+    interval_days = args.days
+    output = args.output
+
+    res: DefaultDict[str, DefaultDict[str, DefaultDict[str, bool]]]
+    res = defaultdict(lambda: defaultdict(lambda: defaultdict(bool)))
```

Comment:
frankly saying, I didn't see that you utilize `defaultdict` features. You're only writing the data, which means, that the usual `dict` will be enough

Reply:
it was my failure. Only for the last `defaultdict(bool)` it makes sense